xbmc/video/videosync/VideoSyncCocoa.cpp

```diff
@@ -33,6 +33,28 @@
 #if defined(TARGET_DARWIN_OSX)
 #include <QuartzCore/CVDisplayLink.h>
 #include "osx/CocoaInterface.h"
+#endif
+
+void CVideoSyncCocoa::OnLostDevice()
+{
+  if (!m_displayLost)
+  {
+    m_displayLost = true;
+    m_lostEvent.Wait();
```

Comment:
CVideoSyncD3D::OnDestroyDevice is a different story. It has to make sure the D3D device is not used any more after the call has completed.